### PR TITLE
workaround for no MKLDNN kernel

### PR DESCRIPTION
**Pull request by pzelazko-intel**
Problem:
When we set environment variable `FLAGS_use_mkldnn` to true, there is an exception thrown, telling that there is no MKLDNN kernel for e.g. sigmoid activation OP. It happens this way, because all activation functions have one GetExpectedKernel method and only few have MKLDNN implementation.

Workaround:
If no MKLDNN kernel can be found, try again with plain CPU one.

Can you see better solutions for this problem?

### CODE REVIEW

**Comment by luotao1** [comment]
@pzelazko-intel The TeamCity log is lost, please push again.

**Comment by luotao1** [comment]
@jacquesqiao Could you help to review this PR, thanks very much!

File: paddle/fluid/framework/operator.cc [hunk 1]
```diff
@@ -635,6 +635,15 @@ void OperatorWithKernel::RunImpl(const Scope& scope,
   VLOG(3) << "expected_kernel_key:" << expected_kernel_key;
 
   auto kernel_iter = kernels.find(expected_kernel_key);
+#ifdef PADDLE_WITH_MKLDNN
+  // workarkound for missing MKLDNN kernel when FLAGS_use_mkldnn env var is set
```
Comment (luotao1) [inline paddle/fluid/framework/operator.cc]: typo: workaround 
Maybe can fix this typo at the same time: https://github.com/PaddlePaddle/Paddle/pull/11544#pullrequestreview-131558634

Comment (pzelazko-intel) [reply paddle/fluid/framework/operator.cc]: done

**Comment by jacquesqiao** [comment]
What is the reason for the framework to throw the exception?

**Comment by luotao1** [comment]
@jacquesqiao The reason is that activation op has `use_mkldnn` attr, so 
https://github.com/PaddlePaddle/Paddle/blob/3d99e2e9df6333c52bca678c06c3ea51344b7894/paddle/fluid/framework/executor.cc#L412-L426
will change the `use_mkldnn` attr to True when set `FLAGS_use_mkldnn`.

However, there a lot of activation ops, but only part of them has MKLDNN kernel. For example, sigmoid activation op doesn't has MKLDNN kernel, thus, it throws the exception.

File: paddle/fluid/framework/operator.cc [hunk 2]
```diff
@@ -633,6 +633,15 @@ void OperatorWithKernel::RunImpl(const Scope& scope,
   VLOG(3) << "expected_kernel_key:" << expected_kernel_key;
 
   auto kernel_iter = kernels.find(expected_kernel_key);
+#ifdef PADDLE_WITH_MKLDNN
+  // workaround for missing MKLDNN kernel when FLAGS_use_mkldnn env var is set
+  if (kernel_iter == kernels.end() &&
+      expected_kernel_key.library_type_ == LibraryType::kMKLDNN) {
+    expected_kernel_key.library_type_ = LibraryType::kPlain;
```
Comment (jacquesqiao) [inline paddle/fluid/framework/operator.cc]: please add a log for the fallback

Comment (pzelazko-intel) [reply paddle/fluid/framework/operator.cc]: done

**Review by jacquesqiao — APPROVED** [review]
LGTM

**Pull request merged**